### ISSUE TITLE
V2.0 Feedback collection

### ISSUE THREAD

**Issue by smstuebe** [smstuebe/xamarin-fingerprint#151]
I have released a preview version of the plugin. 

_**Please provide feedback directly in this issue.**_

**Major changes**
- .netstandard 2.0
- Android X
- replaced custom dialog and Samsung pass with BiometricPrompt
- Fallback authentication for Android
- NoDialog option no longer supported

**Installation**
- Make sure, you follow the install instructions! https://github.com/smstuebe/xamarin-fingerprint/tree/2.0.0
- https://www.nuget.org/packages/Plugin.Fingerprint/2.0.0-alpha.5

**Why prerelease**
Xamarin has not released Android X as stable packages for 1 year... NuGet doesn't allow realeasing stable packages, if the package has pre-release dependencies.

see: 
- https://github.com/xamarin/AndroidSupportComponents/issues/208
- https://github.com/xamarin/AndroidSupportComponents/issues/115

**Known Issues**
- [x] Android: Triggering the authentication after trying to use the alternative authentication, blocks the UI. This is somehow fragment related. I'll have a look, if I can fix it myself.
- [x] Android: Dialog doesn't close if cancellation token gets cancelled.
- [x] Android: 23/6.0/M, GetAvailabilityAsync returns "NoSensor", but should be "Available"
- [ ] Cancel Authentication doesn't work on Android

**Needs to be tested**
- [x] Android: Face unlock
- [x] Android: Alternative authentication that are not numbers (e.g. patterns)

**Other Todos**
- [x] Android: Update to androidx.biometric 1.0.0
- [x] MvvmCross

**Comment by tranb3r** [comment]
Awesome work! No major issue so far for me.
One question: is it possible to test facial unlock on the android emulator? (or does one need a physical pixel 4 for that ?)

**Comment by smstuebe** [comment]
@tranb3r I rarely test with the simulator. I can't really tell if there is/will be support for that in the emulator (google is relatively slow with enhancing the emulator :/). So it seems you have to borrow/buy a pixel device. 
But good point. I'll ask if one of my colleagues/friends own a device using facial unlock. I'll extend the issue with a "things to test" section

**Comment by tranb3r** [comment]
On Android 23/6.0/M, GetAvailabilityAsync now returns "NoSensor" (while it was returning "Available" before).
I see that you're using version rc01 of androidx's biometric. Maybe this has been fixed in the stable 1.0.0 that has been released a few days ago ?

**Comment by smstuebe** [comment]
@tranb3r yes. I tried to update to 1.0.0 yesterday, but unfortunately it depends on 1.1.0 versions of other android x packages that are not released, yet :( You can follow the discussion if you click on the linked Issue 208)

**Comment by joshykautz** [comment]
Got a little time to play around with it - nothing giving me trouble so far! Thanks for the prelease; all your hard work is greatly appreciated! Hope to be able to do my part to help!

**Comment by srqdev** [comment]
Running the test app on a Galaxy S10. Two things, first the facial recognition works and is very quick. But second, I would like to be able to select between facial and finger recognition, I have facial recognition turned on for my device and this works with the dialog at the bottom. If I turn off facial recognition, the dialog moves into the middle and works correctly. For my app I would like to only recognize fingerprints i.e. a positive action to log in and not just my face happens to be there. Also can we have a bit more configurability in dialog to be able to change the icon. Everything else seems good so far.

**Comment by tranb3r** [comment]
Xamarin.AndroidX.Biometric 1.0.0-preview03 nuget has been published, along with 1.1.0-preview03 dependencies.

**Comment by smstuebe** [comment]
Updated Dependency to official AndroidX nuget:
https://www.nuget.org/packages/Plugin.Fingerprint/2.0.0-alpha.3

**Comment by tranb3r** [comment]
The issue on Android 23 that I mentionned before is fixed in alpha4. Thanks for that !

**Comment by smstuebe** [comment]
> The issue on Android 23 that I mentionned before is fixed in alpha4. Thanks for that !

That's awesome. Thx for testing.

**Comment by lhughey** [comment]
I haven't tested this out (yet). I'll begin doing so this weekend. 

I wanted to thank you for the time and effort that you put into this component. There are way too many apps that depend on your hard work. I made a donation. Its not enough for the value you provide but consider it a token of my appreciation.

**Comment by pzimmermaninmo** [comment]
Tested latest 2.0.0-alpha.5 release on Android (within Xamarin forms app) and everything seems to be working fine.  

However, I tried setting the current activity using the **current activity plugin** like:
` CrossFingerprint.SetCurrentActivityResolver(() => CrossCurrentActivity.Current.Activity);`
And it would either throw an exception saying "Are you sure youre using the androidx.migration library" (which I am) or just wouldn't display at all.  Not sure if thats specific to your library or something with xamarin/android x itself.  One thing to note here, I am using the Rg.Plugins.Popup to ask the user if they would like to use biometrics to login to my app before calling **AuthenticateAsync**.

That said, using the **MainActivity** like this, as shown in the sample app worked:
`CrossFingerprint.SetCurrentActivityResolver(() => this);`

Like everyone else has said, I appreciate all your work on getting the androidx library working with this.  Previously I was rolling my own Biometric service and was dreading having to create a custom biometric prompt for android 6-8.  Why Google didn't standardize that in the API like iOS is beyond me.

**Comment by smstuebe** [comment]
@lhughey thanks alot :)
@pzimmermaninmo
I assume you are hitting this exception? Can you post the exact exception Message? Do you have a sample app where this error occurs?

**Comment by pzimmermaninmo** [comment]
That was the exception.  I figured out the problem.  I forgot to add this to my **MainActivity**:
`CrossCurrentActivity.Current.Init(this, bundle);`

Carry On.

**Comment by smstuebe** [comment]
@pzimmermaninmo ok. But maybe I can make it more clear. I'll add a check for null and hint the user to his possible mistake better.

**Comment by marceloarchiza** [comment]
@smstuebe I tested on Android, and the FaceID did not work. See the following video:
https://www.youtube.com/watch?v=3A3p-DtVcmg&feature=youtu.be
Tested on Samsung S10 (Android 9.0)

**Comment by marceloarchiza** [comment]
Just an Update. I tested on a Pixel 4 (Android 10) and it worked as expected. I believe this library is not expected to support Samsung Android 9 Face Recognition, right? Just the phones that use the android 10 biometrics?

**Comment by smstuebe** [comment]
Hmm there is one more option regarding face unlock I want to play around with. But it's not like there is a list of methods I can allow. 
Ofc: if the vendor does not implement the API well, I can't do anything :(

>  You are a true MVP

I'm not an MS MVP anymore ^^

**Comment by vineet1810** [comment]
can someone send code for faceid authentication ?

**Comment by noxe** [comment]
Hi,

i currently use V2.0 of the fingerprint plugin - everything works really fine - thank you for that!

However i have one issue - i use Media Plugin to take Photos - when returning to the app OnResume is called on Android - in this case (i want fingerprint authentication in generell when returning to the app) - the authentication craches with this error. I there a bug in the plugin - or do i need to configure something different here when returning from the media picker?

thank you!!

System.InvalidOperationException: 'Expected current activity to be 'AndroidX.Fragment.App.FragmentActivity' but was 'Plugin.Media.MediaPickerActivity'. You need to use AndroidX. Have you installed Xamarin.AndroidX.Migration in your Android App project!?'

 	0x8A in Plugin.Fingerprint.FingerprintImplementation.NativeAuthenticateAsync at d:\a\1\s\src\Plugin.Fingerprint\Platforms\Android\FingerprintImplementation.cs:74,17	C#
 	0x33 in System.Runtime.CompilerServices.AsyncTaskMethodBuilder<Plugin.Fingerprint.Abstractions.FingerprintAuthenticationResult>.Start<Plugin.Fingerprint.FingerprintImplementation.<NativeAuthenticateAsync>d__4> at /Users/builder/jenkins/workspace/archive-mono/2019-08/android/release/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:471,17	C#
 	0x2F in Plugin.Fingerprint.FingerprintImplementation.NativeAuthenticateAsync	C#
 	0xC2 in Plugin.Fingerprint.Abstractions.FingerprintImplementationBase.AuthenticateAsync at d:\a\1\s\src\Plugin.Fingerprint\Abstractions\FingerprintImplementationBase.cs:24,13	C#
 	0x33 in System.Runtime.CompilerServices.AsyncTaskMethodBuilder<Plugin.Fingerprint.Abstractions.FingerprintAuthenticationResult>.Start<Plugin.Fingerprint.Abstractions.FingerprintImplementationBase.<AuthenticateAsync>d__0> at /Users/builder/jenkins/workspace/archive-mono/2019-08/android/release/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:471,17	C#
 	0x37 in Plugin.Fingerprint.Abstractions.FingerprintImplementationBase.AuthenticateAsync	C#
>	0xCB in VenDoc.App.RaiseAuthentication at E:\Entwicklung\VenDoc_Mobile\VenDoc\VenDoc\App.xaml.cs:110,17	C#
 	0x33 in System.Runtime.CompilerServices.AsyncTaskMethodBuilder<bool>.Start<VenDoc.App.<RaiseAuthentication>d__9> at /Users/builder/jenkins/workspace/archive-mono/2019-08/android/release/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:471,17	C#
 	0x23 in VenDoc.App.RaiseAuthentication	C#
 	0x1A in VenDoc.App.EnterForeground at E:\Entwicklung\VenDoc_Mobile\VenDoc\VenDoc\App.xaml.cs:57,13	C#
 	0x33 in System.Runtime.CompilerServices.AsyncVoidMethodBuilder.Start<VenDoc.App.<EnterForeground>d__4> at /Users/builder/jenkins/workspace/archive-mono/2019-08/android/release/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:84,17	C#
 	0x2A in VenDoc.App.EnterForeground	C#
 	0x1C in VenDoc.App.OnResume at E:\Entwicklung\VenDoc_Mobile\VenDoc\VenDoc\App.xaml.cs:100,17	C#

**Comment by bijoyjacob** [comment]
@smstuebe I have XF application, which uses Plugin.Fingerprint v2.1.1 with all AndroidX libraries at tip, except Xamarin.AndroidX.Browser(my project uses v1.0 as v1.2 throws error). However, after the Fingerprint Dialog been shown and either Press Cancel or Place Finger after that it dismisses the dialog, but mobile app gets hung. Not sure of anyone came across this issue. Note - My XF application has a Webview which loads a website and planning to add biometric to it.

Below is the Packages.Config
<package id="Microsoft.AppCenter" version="3.1.0" targetFramework="monoandroid90" />
  <package id="Microsoft.AppCenter.Analytics" version="3.1.0" targetFramework="monoandroid90" />
  <package id="Microsoft.AppCenter.Crashes" version="3.1.0" targetFramework="monoandroid90" />
  <package id="Plugin.CurrentActivity" version="2.1.0.4" targetFramework="monoandroid10.0" />
  <package id="Plugin.Fingerprint" version="2.1.1" targetFramework="monoandroid90" />
  <package id="Xam.Plugin.Connectivity" version="3.2.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Arch.Core.Common" version="1.1.1.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Arch.Core.Runtime" version="1.1.1.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Arch.Lifecycle.Common" version="1.1.1.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Arch.Lifecycle.LiveData" version="1.1.1.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Arch.Lifecycle.LiveData.Core" version="1.1.1.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Arch.Lifecycle.Runtime" version="1.1.1.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Arch.Lifecycle.ViewModel" version="1.1.1.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Animated.Vector.Drawable" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Annotations" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.AsyncLayoutInflater" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Collections" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Compat" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.CoordinaterLayout" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Core.UI" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Core.Utils" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.CursorAdapter" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.CustomTabs" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.CustomView" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Design" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.DocumentFile" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.DrawerLayout" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Fragment" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Interpolator" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Loader" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.LocalBroadcastManager" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Media.Compat" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Print" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.SlidingPaneLayout" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.SwipeRefreshLayout" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Transition" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.v4" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.v7.AppCompat" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.v7.CardView" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.v7.MediaRouter" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.v7.Palette" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.v7.RecyclerView" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.Vector.Drawable" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.VersionedParcelable" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.Android.Support.ViewPager" version="28.0.0.3" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Activity" version="1.1.0" targetFramework="monoandroid10.0" />
  <package id="Xamarin.AndroidX.Annotation" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.AppCompat" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.AppCompat.Resources" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Arch.Core.Common" version="2.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Arch.Core.Runtime" version="2.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.AsyncLayoutInflater" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Biometric" version="1.0.1" targetFramework="monoandroid10.0" />
  <package id="Xamarin.AndroidX.Browser" version="1.0.0" targetFramework="monoandroid10.0" />
  <package id="Xamarin.AndroidX.CardView" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Collection" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.CoordinatorLayout" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Core" version="1.2.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.CursorAdapter" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.CustomView" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.DocumentFile" version="1.0.1" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.DrawerLayout" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Fragment" version="1.2.3" targetFramework="monoandroid10.0" />
  <package id="Xamarin.AndroidX.Interpolator" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Legacy.Support.Core.UI" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Legacy.Support.Core.Utils" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Legacy.Support.V4" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Lifecycle.Common" version="2.2.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Lifecycle.LiveData" version="2.2.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Lifecycle.LiveData.Core" version="2.2.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Lifecycle.Runtime" version="2.2.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Lifecycle.ViewModel" version="2.2.0" targetFramework="monoandroid10.0" />
  <package id="Xamarin.AndroidX.Lifecycle.ViewModelSavedState" version="2.2.0" targetFramework="monoandroid10.0" />
  <package id="Xamarin.AndroidX.Loader" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.LocalBroadcastManager" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Media" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.MediaRouter" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Migration" version="1.0.4" targetFramework="monoandroid10.0" />
  <package id="Xamarin.AndroidX.MultiDex" version="2.0.1" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Palette" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Print" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.RecyclerView" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.SavedState" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.SlidingPaneLayout" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.SwipeRefreshLayout" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.Transition" version="1.3.1" targetFramework="monoandroid10.0" />
  <package id="Xamarin.AndroidX.VectorDrawable" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.VectorDrawable.Animated" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.VersionedParcelable" version="1.1.0" targetFramework="monoandroid90" />
  <package id="Xamarin.AndroidX.ViewPager" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Build.Download" version="0.10.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Firebase.Common" version="71.1610.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Firebase.Iid" version="71.1710.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Firebase.Iid.Interop" version="71.1601.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Firebase.Measurement.Connector" version="71.1701.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Firebase.Messaging" version="71.1740.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Forms" version="4.5.0.617" targetFramework="monoandroid10.0" />
  <package id="Xamarin.Google.Android.Material" version="1.0.0" targetFramework="monoandroid90" />
  <package id="Xamarin.Google.AutoValue.Annotations" version="1.6.5" targetFramework="monoandroid90" />
  <package id="Xamarin.Google.Guava.ListenableFuture" version="1.0.0.2" targetFramework="monoandroid90" />
  <package id="Xamarin.GooglePlayServices.Ads.Base" version="71.1720.0" targetFramework="monoandroid90" />
  <package id="Xamarin.GooglePlayServices.Base" version="71.1610.0" targetFramework="monoandroid90" />
  <package id="Xamarin.GooglePlayServices.Basement" version="71.1620.0" targetFramework="monoandroid90" />
  <package id="Xamarin.GooglePlayServices.Gass" version="71.1720.0" targetFramework="monoandroid90" />
  <package id="Xamarin.GooglePlayServices.Stats" version="71.1601.0" targetFramework="monoandroid90" />
  <package id="Xamarin.GooglePlayServices.Tasks" version="71.1601.0" targetFramework="monoandroid90" />

**Comment by katanza12** [comment]
I have the same problem. For now I downgraded the plugin at the last version before 2.0.0 (that implement the FragmentActivity) and it works. But we need a different solution for the future

**Comment by Ayesh02** [comment]
Hi,
I am trying to implement face id authentication using this library but for some reason it is not working for Android. It could not detect the face id but only fingerprint. Please assist where i might have gone wrong.

nuget version: 2.1.1
Installed androidx.migration as well.
Granted use_biometric permission as well

device os: Android 10
device name: Samsung Galaxy Note 9

**Comment by Ayesh02** [comment]
Hi,
I am trying to implement face id authentication using this library. I am trying to determine auth type based on this method:
await CrossFingerprint.Current.GetAuthenticationTypeAsync()

However, i am always getting the result as Fingerprint instead of Face. Can i know why?

device os: Android 9
device name: Samsung Galaxy Note 9

**Comment by JohnBirju** [comment]
Dear  @smstuebe,
In our Xamarin forms application we have provided biometric login support PFB details for that.
Nudget package name : 
Version:
Method/ API : 
Platform:
Issue description: above used API does not return Face id, each time it returning Touch id.
Ex

**Comment by JohnBirju** [comment]
Dear @Ayesh02 ,
In our application we are also facing same issue ( on Android above mentioned method always return touch id, even the Face id enabled on device) could you please help me, if you have the solution for that issue.

Thanks and regards: John.

**Comment by AnEnglishmanInNorway** [comment]
Hi
First of all - I have been using Plugin.Fingerprint for some years and am most happy to be able to rely on it, thank you! I am now trying to upgrade to 2.1.1 and having a little trouble with one device, on which more later. 
First, my environment:
Platforms on which I currently have my app operational are:
* Xamarin UWP on Windows 10
* Xamarin Android 6.0 on Samsung Galaxy S5
* Xamarin Android 10.0 on Samsung Galaxy S10 and Sony Xperia XZ2 Compact

I use Visual Studio 2019 Community Edition on Windows 10 as my development environment.
I currently have Android 10.0 (Q, API 29) as my target framework when I build, but set a minimum version as Android 6.0 (Marshmallow, API 23)
For Android I test first using Visual Studio's Android emulators configured for the various API levels (and with a fingerprint sensor enabled). Then I attach the physical phones which are most important to me, usually just the two I have in use myself, and check them using USB debugging.
In all cases the OS and the Nuget packages on which I rely are the latest as of today's date (I have updated the app regularly as updates have become available)
The one exception to my "keep everything up to date" rule has been Plugin.Fingerprint, where I have stayed with 1.4.9. Now I am attempting to update to 2.1.1. Things are mostly good after the update:
- the app runs ok on the emulators for API 23 and API 29
- the app runs ok on a physical Sony Xperia XZ2
BUT
- the fingerprint reading on a physical Galaxy S5 fails, because GetAvailabilityAsync thinks there is no sensor.

Perhaps somebody might have some idea of what might be the issue. I know Samsung have been a bit special in the past with the WRITE_USE_APP_FEATURE_SURVEY permission they require (and which I have set in my manifest). I have not yet had time to test on a physical S10 to see if the problem is there too and will be away from my environment for a while. Right now I have given up and gone back to 1.4.9, but I thought you might be interested!
